var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -683,16 +683,16 @@ def cmake_args(self):
                 # hence the test to see if the version starts with "flang".
                 targets.append('CppBackend')
 
-            if 'x86' in spec.architecture.target.lower():
+            if 'x86' in spec.target.family:
                 targets.append('X86')
-            elif 'arm' in spec.architecture.target.lower():
+            elif 'arm' in spec.target.family:
                 targets.append('ARM')
-            elif 'aarch64' in spec.architecture.target.lower():
+            elif 'aarch64' in spec.target.family:
                 targets.append('AArch64')
-            elif 'sparc' in spec.architecture.target.lower():
+            elif 'sparc' in spec.target.family:
                 targets.append('Sparc')
-            elif ('ppc' in spec.architecture.target.lower() or
-                  'power' in spec.architecture.target.lower()):
+            elif ('ppc' in spec.target.family or
```

Comment:
```suggestion
            elif (spec.target.family == 'ppc64' or spec.target.family == 'ppc64le' or
                  spec.target.family == 'ppc' or spec.target.family == 'ppcle'):
```
but if you want to avoid `ppc` and `ppcle` it should be fine too.